crates/types/src/nodes_config.rs

```diff
@@ -373,9 +376,9 @@ impl StorageState {
         }
     }
 
-    /// Empty nodes are automatically excluded from node sets.
+    /// Empty nodes are considered not part of the nodeset and they'll never join back.
     pub fn empty(&self) -> bool {
-        matches!(self, StorageState::Provisioning | StorageState::Disabled)
+        matches!(self, StorageState::Disabled)
```

Comment:
Classifying `Provisioning` nodes as not empty makes them to be included in the `EffectiveNodeSet`, right? So they would be considered for various tasks, like reading, sealing, trimming etc. Is this intended? I assume that this is right now not a problem because of the updated `logserver_candidate_filter` which excludes `Provisioing` nodes from becoming part of a `NodeSet`.

Reply:
Correct. It wouldn't be a problem even if I remove the commented block of the filter. That's because still try and select enough "writeable" nodes in nodesets, but if we can't we will have those nodes in nodesets.

So, the problem may happen only when a bunch of provisioning nodes are added to a nodeset and they never transition to ReadWrite, then we can't seal that loglet.